### PR TITLE
initial data fed work

### PR DESCRIPTION
**Pull request by polandll**
incomplete - need to finish

### CODE REVIEW

File: modules/developers-guide/pages/data_federation.adoc [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+A Production object tracks everything needed to make a Movie including shooting location, vendors, and more.
+Talent: the people working on a Movie are the Talent, including actors, directors, and so on.
+These three domains are owned by three separate engineering teams responsible for
+their own data sources, business logic, and corresponding microservices.
+In an unfederated implementation, we would have this simple Schema and Resolvers
+owned and implemented by the Studio API team.
+The GraphQL Framework would take in queries from clients and orchestrate the
+calls to the resolvers in a breadth-first traversal.
+
+SEE DIAGRAM
```
Comment (dougwettlaufer) [inline modules/developers-guide/pages/data_federation.adoc:23]: Where's the diagram for this?

File: modules/developers-guide/pages/data_federation.adoc [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+12. (readerByNameAndUserid) Look up a reader by name and user_id (primary key). _Reader_
+    All 3 of these queries will allow you to get all the books reviewed by a reader.
+
+13. (orders) Get all orders. _Order_
+13a. (order) Get order by checkout_id. _Order_
+13b. (orderById) Look up an order by checkout_id (partition key). _Order_
+    This will give you all books in the order.
+
+14. CAN'T DO?(libCollByType) Look up a library collection by type (partition key, partial primary key). _LibCollection_
+15. CAN'T DO(libCollByLibid) Look up a library collection by lib_id (partition key, partial primary key). _LibCollection_
```
Comment (dougwettlaufer) [inline modules/developers-guide/pages/data_federation.adoc:77]: Is there an answer yet on whether you can or can't do this?

File: modules/developers-guide/pages/data_federation.adoc [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+16. (libCollByTypeAndLibid) Look up a library collection by type and lib_id (primary key). _LibCollection_
+17. (libCollByName) Look up a library collection by name (non-PK column, index). _LibCollection_
+
+20.(reviewsByBook) Look up a book and find all reviews and return review+book info. _union_ of _Book_ and _Reader_
+    Find a book, check all readers for a review of the book by Review.bookTitle.
+21. (orderByBook) Look up a book and find all the readers who have checked it out. _union_ of _Book_ and _Order_
+    Find a book, look up the orders, get the readers.
+22. THIS IS readerCONTAINS(fullReview) For a reader's review, get all the book information (non-PK column collection, index)
+Requires all review information. _Reader_
+23. (??) Look at a reader and get all books read. _union_ of _Book_ and _Reader_
```
Comment (dougwettlaufer) [inline modules/developers-guide/pages/data_federation.adoc:87]: What's up with this one?

File: modules/developers-guide/pages/old_data_fed_text.adoc [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+Similar to the type name, the fields can be identified with an alternative
+Cassandra column name using `@cql_column(name: "book_title")`.
+This directive is not required.
+
+In the Book type, there are additional `@cql_column` parameters defined to
+identify the primary key of the type.
+Define the partition key using `@cql_column(partitionKey: true)` for all fields
+that comprise the partition key.
+Define the clustering columns using `@cql_column(clusteringOrder: ASC)`, for all
+the fieldsthat comprise the clustering columns, if desired.
```
Comment (dougwettlaufer) [inline modules/developers-guide/pages/old_data_fed_text.adoc:35]:
```suggestion
the fields that comprise the clustering columns, if desired.
```

File: modules/developers-guide/pages/old_data_fed_text.adoc [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+====
+If no field has a `partitionKey: true`, but the first field is of data type
+ID, Uuid, or TimeUuid, then that field is used as the partition key in the database.
+====
+
+A standard GraphQL type feature is displayed in these types, the required fields
+denoted by `String!` and `Uuid!` that include the exclamation mark.
+*** MAKES THEM NON-NULL FIELDS - PARTITION KEY MUST BE NON-NULL!
+
+he `@cql_index` is discussed in xref:graphql-first-using.adoc#_create_indexes[Create indexes].
```
Comment (dougwettlaufer) [inline modules/developers-guide/pages/old_data_fed_text.adoc:52]:
```suggestion
The `@cql_index` is discussed in xref:graphql-first-using.adoc#_create_indexes[Create indexes].
```

File: modules/developers-guide/pages/apollo-federation.adoc [hunk 6]
```diff
@@ -1,6 +1,296 @@
...
+include::example$graphql/1orders.js[]
+----
+--
+====
 
 == Installing Apollo gateway
 
-== GraphQL schema required
+Let's look at a simple Javascript that runs the Apollo gateway:
+Let's break down this script down as well, into three parts.
```
Comment (dougwettlaufer) [inline modules/developers-guide/pages/apollo-federation.adoc:187]:
```suggestion
Let's break this script down as well, into three parts.
```

**Review by dougwettlaufer — APPROVED** [review]

File: modules/developers-guide/pages/data_federation.adoc (hunk 1 above)
Comment (polandll) [reply modules/developers-guide/pages/data_federation.adoc]: I thought this page had been removed.

File: modules/developers-guide/pages/data_federation.adoc (hunk 2 above)
Comment (polandll) [reply modules/developers-guide/pages/data_federation.adoc]: Removed page.

File: modules/developers-guide/pages/data_federation.adoc (hunk 3 above)
Comment (polandll) [reply modules/developers-guide/pages/data_federation.adoc]: Removed page.

File: modules/developers-guide/pages/old_data_fed_text.adoc (hunk 4 above)
Comment (polandll) [reply modules/developers-guide/pages/old_data_fed_text.adoc]: Removed page.

File: modules/developers-guide/pages/old_data_fed_text.adoc (hunk 5 above)
Comment (polandll) [reply modules/developers-guide/pages/old_data_fed_text.adoc]: Removed page.

**Pull request merged**